spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/support/tagsprovider/GatewayTagsProvider.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gateway.support.tagsprovider;
+
+import io.micrometer.core.instrument.Tags;
+
+import org.springframework.web.server.ServerWebExchange;
+
+/**
+ * @author Ingyu Hwang
+ */
+public interface GatewayTagsProvider {
```

Comment:
I wonder if making this extend `Function<ServerWebExchange, Tags>` would help with the composite.